### ISSUE TITLE
Add examples to dostrings

### ISSUE THREAD

**Issue by bmu** [pvlib/pvlib-python#62]
We should decide whether we want to add examples to the docstrings of the functions or not.
While (I think) examples are not included in every pandas object, they are in numpy. [numpydoc](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#docstring-standard) has an *optional* section for examples.

I would like to include examples and also I would like to use doctests.
For the moment we have some examples, but some doctests fail 
(see output of `nosetests -v -w pvlib --with-coverage --cover-package=pvlib --with-doctest`).
I once had a discussion with pandas developers (see pydata/pandas#3439) and they didn't see their examples as tests. 
However for pvlib, `doctest` may have an additional benefit for testing purposes and also it tests if our documentation is up to date with the state of the objects we document.

**Comment by wholmgren** [comment]
We definitely need more examples in the docs. I don't know if we need an example for absolutely everything, though. In any case, we probably will want to change the documentation formatting to 1 page per function.

I don't have any experience using ``doctest``, but I'm ok with trying it out. I would say that we should not rely on ``doctest`` for anything other than ensuring that the docs are up to date with the library, and that anything that is in a doctest should be in the main tests.

**Comment by mikofski** [comment]
Any decision on this?

**Comment by wholmgren** [comment]
@mikofski I support adding brief examples to the function docstrings where they would be useful. Personally, I find documentation pages structured like [ModelChain](https://pvlib-python.readthedocs.io/en/latest/modelchain.html) and [Clearsky](https://pvlib-python.readthedocs.io/en/latest/clearsky.html) to be more useful, so that's where I've focused my documentation efforts. But I encourage people that want small examples for each function to take on the effort. So I lean toward leaving this generic issue open. What do you think?

**Comment by abhinav9414** [comment]
can I try to do this sir.

**Comment by cwhanse** [comment]
> can I try to do this sir.

Certainly. Are you somewhat familiar with PV system modeling? That would help with forming examples.

It would be best to take this up in small pieces. Maybe the file `iam.py` is a place to begin.

**Comment by abhinav9414** [comment]
ok ill start with iam.py file thanks for the boast.

**Comment by dgowdaan-cmyk** [comment]
Hi, I'd like to add docstring examples to functions in iam.py that are currently missing them. I noticed that ashrae, physical, martin_ruiz, martin_ruiz_diffuse, interp, sapm, and schlick don't have an Examples section yet. I'd like to start with ashrae, physical, and martin_ruiz. Is this still welcome as a contribution?

**Comment by dgowdaan-cmyk** [comment]
Update: I've opened #2797 to add an Examples section to iam.ashrae — currently open for review.

I'll continue with iam.physical and iam.martin_ruiz next, as mentioned above. Will open separate PRs for each to keep the reviews focused and easy to merge.
